### PR TITLE
ci.yml: Add macOS on ARM64 to the testing

### PR DESCRIPTION
**Pull request by cclauss**
https://github.com/federicocarboni/setup-ffmpeg/issues/21#issuecomment-2480376403

### CODE REVIEW

**Comment by henryruhs** [comment]
@cclauss  Thanks for the pull request, it is indeed time to finally test latest M1 series.

1. Remove macos-13
2. Use matrix.os checks
3. Do not use != statements
4. Use ffmpeg 6 over 3

```
if: matrix.os == 'ubuntu-latest'
if: matrix.os == 'windows-latest'
if: matrix.os == 'macos-latest'
```

**Comment by cclauss** [comment]
2. Is not the recommendation of the GHA docs but I will make the change.
    * https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/store-information-in-variables#detecting-the-operating-system

4. Is a bit confusing.  The brew formula currently defaults to ffmpeg@7.1.  Should we downgrade to ffmpeg@6?
    * https://formulae.brew.sh/formula/ffmpeg

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,9 +27,9 @@ jobs:
   steps:
   - name: Checkout
     uses: actions/checkout@v4
-  - name: Set up FFmpeg
+  - name: Set up <<<<<<< patch-1
```
Comment (cclauss) [inline .github/workflows/ci.yml:30]:
```suggestion
 - name: Set up FFmpeg
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -22,14 +22,14 @@ jobs:
...
   runs-on: ${{ matrix.os }}
   steps:
   - name: Checkout
     uses: actions/checkout@v4
-  - name: Set up FFmpeg
-    uses: FedericoCarboni/setup-ffmpeg@v3
-  - name: Set up Python 3.12
+ - name: Set up FFmpeg
+    uses: AnimMouse/setup-ffmpeg@v1
+  - name: Set up Python 3.2
```
Comment (cclauss) [inline .github/workflows/ci.yml:32]:
```suggestion
  - name: Set up Python 3.12
```

**Comment by henryruhs** [comment]
Using `AnimMouse/setup-ffmpeg@v1` just works, awesome. Thanks for your time and efforts.

**Pull request merged**